subprojects/GOPATH/src/github.com/microsoft/hivedscheduler/pkg/algorithm/hived_algorithm.go

```diff
@@ -855,6 +966,23 @@ func (h *HivedAlgorithm) findPhysicalGpuInChain(
 	return nil
 }
 
+// updateVirtualClusterCapacity calculates the remaining capacity in a VC in case of failures.
+// In the best case, a VC can use all the remaining quota as long as the physical cluster still has
+// as much capacity after counting failures (assuming other VCs will not compete for these GPUs).
+// In the worst case, a VC's capacity is the remaining quota minus the number of failed GPUs
+// (assuming all the normal GPUs are used by other VCs).
+func (h *HivedAlgorithm) updateVirtualClusterCapacity(vc *api.ClusterCapacity) {
+	vc.RemainingCapacityBestCase = h.clusterCapacity.Total - h.clusterCapacity.GuaranteedUsage - h.failedGpuNum
+	if vc.Total-vc.GuaranteedUsage < vc.RemainingCapacityBestCase {
+		vc.RemainingCapacityBestCase = vc.Total - vc.GuaranteedUsage
+	}
+
+	vc.RemainingCapacityWorstCase = vc.Total - vc.GuaranteedUsage - h.failedGpuNum
```

Comment:
what if current vc shares gpus with another vc and all shared gpus are used by other?